protocol/contracts/lease/src/contract/state/paid/v8.rs

```diff
@@ -0,0 +1,98 @@
+use serde::Deserialize;
+
+use currencies::LeaseGroup;
+use currency::SymbolSlice;
+use dex::{Account, CoinVisitor, IterNext, IterState, SwapTask};
+use finance::coin::CoinDTO;
+use oracle_platform::OracleRef;
+use sdk::cosmwasm_std::{Env, QuerierWrapper};
+use timealarms::stub::TimeAlarmsRef;
+
+use crate::{
+    api::{query::StateResponse, LeasePaymentCurrencies},
+    contract::{
+        state::{
+            resp_delivery::{ForwardToDexEntry, ForwardToDexEntryContinue},
+            SwapClient, SwapResult,
+        },
+        v8::Lease,
+    },
+    error::ContractResult,
+    event::Type,
+    finance::ReserveRef,
+};
+
+use super::{transfer_in::TransferIn as TransferIn_v9, Active as Active_v9};
+
+pub(crate) type DexState = dex::StateLocalOut<
+    TransferIn,
+    LeasePaymentCurrencies,
+    SwapClient,
+    ForwardToDexEntry,
+    ForwardToDexEntryContinue,
+>;
+
+#[derive(Deserialize)]
+pub(crate) struct Active {
+    lease: Lease,
+}
+
+impl Active {
+    pub(crate) fn migrate(self, reserve: ReserveRef) -> Active_v9 {
+        Active_v9::new(self.lease.migrate(reserve))
+    }
+}
+
+#[derive(Deserialize)]
+pub(crate) struct TransferIn {
+    lease: Lease,
+}
+
+impl TransferIn {
+    pub(crate) fn migrate(self, reserve: ReserveRef) -> TransferIn_v9 {
+        TransferIn_v9::new(self.lease.migrate(reserve))
+    }
+}
+
+impl SwapTask for TransferIn {
+    type OutG = LeaseGroup;
+    type Label = Type;
+    type StateResponse = ContractResult<StateResponse>;
+    type Result = SwapResult;
+
+    fn label(&self) -> Self::Label {
+        unreachable!()
```

Comment:
Just as a side note, this heavily violates OO principles, specifically the Liskov substitution principle (L in SOLID).

While this is just a mock/stub used for migration, it might be good to rethink the trait bounds or at least the places where they are put. This should of course happen as future work.

Reply:
I have tried to go without this implementation but my effort ended up stuck on a bigger refactoring related to relaxing trait bounds ...  that is out of the scope of this PR.
In addition, as a true believer in these principles, I do not have any concerns providing such impl-s as they are required only technically for satisfying the compiler's requirements and are *not* used in practice.